src/radiation/radiation.cpp

```diff
@@ -304,27 +305,115 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+
+    const std::string recon_fixup_strategy_str =
+        pin->GetOrAddString("radiation", "recon_fixup_strategy", "bounds");
+    ReconFixupStrategy recon_fixup_strategy;
+    if (recon_fixup_strategy_str == "none") {
+      recon_fixup_strategy = ReconFixupStrategy::none;
+    } else if (recon_fixup_strategy_str == "bounds") {
+      recon_fixup_strategy = ReconFixupStrategy::bounds;
+    } else {
+      PARTHENON_FAIL("\"radiation/recon_fixup_strategy\" option unrecognized!");
+    }
+    params.Add("recon_fixup_strategy", recon_fixup_strategy);
+
+    //    std::string solver = pin->GetOrAddString("radiation", "riemann", "hll");
+    //    riemann::solver rs = riemann::solver::HLL;
+    //    if (solver == "llf") {
+    //      rs = riemann::solver::LLF;
+    //    } else if (solver == "hll") {
+    //      rs = riemann::solver::HLL;
+    //    } else {
+    //      PARTHENON_THROW("Invalid Riemann Solver option. Choose from [llf, hll]");
+    //    }
+    //    params.Add("RiemannSolver", rs);
+
+    const std::string src_solver_name =
+        pin->GetOrAddString("radiation", "src_solver", "fourd");
+    SourceSolver src_solver;
+    if (src_solver_name == "zerod") {
+      src_solver = SourceSolver::zerod;
+    } else if (src_solver_name == "oned") {
+      src_solver = SourceSolver::oned;
+    } else if (src_solver_name == "fourd") {
+      src_solver = SourceSolver::fourd;
+    } else {
+      PARTHENON_FAIL("\"radiation/src_solver\" option unrecognized!");
+    }
+    params.Add("src_solver", src_solver);
+
+    const bool src_use_oned_backup =
+        pin->GetOrAddBoolean("radiation", "src_use_oned_backup", false);
+    params.Add("src_use_oned_backup", src_use_oned_backup);
+
+    Real src_rootfind_eps = pin->GetOrAddReal("radiation", "src_rootfind_eps", 1.e-8);
+    params.Add("src_rootfind_eps", src_rootfind_eps);
+
+    Real src_rootfind_tol = pin->GetOrAddReal("radiation", "src_rootfind_tol", 1.e-12);
+    params.Add("src_rootfind_tol", src_rootfind_tol);
+
+    int src_rootfind_maxiter =
+        pin->GetOrAddInteger("radiation", "src_rootfind_maxiter", 100);
+    params.Add("src_rootfind_maxiter", src_rootfind_maxiter);
+
+    std::string oned_fixup_strategy_str =
+        pin->GetOrAddString("radiation", "oned_fixup_strategy", "none");
+    OneDFixupStrategy oned_fixup_strategy;
+    if (oned_fixup_strategy_str == "none") {
+      oned_fixup_strategy = OneDFixupStrategy::none;
+    } else if (oned_fixup_strategy_str == "ignore_dJ") {
+      oned_fixup_strategy = OneDFixupStrategy::ignore_dJ;
+    } else if (oned_fixup_strategy_str == "ignore_all") {
+      oned_fixup_strategy = OneDFixupStrategy::ignore_all;
+    } else {
+      PARTHENON_FAIL("radiation/oned_fixup_strategy has an invalid entry!");
+    }
+    params.Add("oned_fixup_strategy", oned_fixup_strategy);
+
     int ndim = 3;
     // if (pin->GetInteger("parthenon/mesh", "nx3") > 1) ndim = 3;
     // else if (pin->GetInteger("parthenon/mesh", "nx2") > 1) ndim = 2;
```

Comment:
I guess we don't need this? delete?

Reply:
Actually I think we can reintroduce this since `ndim` is only used for some appropriately guarded reconstruction variables. Not sure why this was ever defaulted to `ndim=3`.